msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/crosscloud/TestCase1420484.java

```diff
@@ -0,0 +1,134 @@
+
+import java.util.Arrays;
+import java.util.Collection;
+
+// Acquiring token for cross cloud guest account (Msal Only)
+// https://identitydivision.visualstudio.com/DefaultCollection/IDDP/_workitems/edit/1420484
+@RunWith(Parameterized.class)
+public class TestCase1420484 extends AbstractGuestAccountMsalUiTest {
+
+    private final String mGuestHomeAzureEnvironment;
+
+    public TestCase1420484(final String name, final String guestHomeAzureEnvironment) {
+        mGuestHomeAzureEnvironment = guestHomeAzureEnvironment;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection guestHomeAzureEnvironment() {
+        return Arrays.asList(new Object[][]{
+                {"AZURE_US_GOV", LabConstants.GuestHomeAzureEnvironment.AZURE_US_GOV},
+                {"AZURE_CHINA_CLOUD", LabConstants.GuestHomeAzureEnvironment.AZURE_CHINA_CLOUD},
+        });
+    }
+
+    /**
+     * Tests Acquiring token for Cross cloud Guest account without broker.
+     */
+    @Test
+    public void test_1420484() throws Throwable {
+        final String userName = mGuestUser.getHomeUpn();
+        final String password = LabGuestAccountHelper.getPasswordForGuestUser(mGuestUser);
+
+        // Handler for Interactive auth call
+        OnInteractionRequired interactionHandler = () -> {
+            PromptHandlerParameters promptHandlerParameters = null;
+            ((IApp) mBrowser).handleFirstRun();
+            promptHandlerParameters = PromptHandlerParameters.builder()
+                    .prompt(PromptParameter.SELECT_ACCOUNT)
+                    .loginHint(userName)
+                    .staySignedInPageExpected(true)
+                    .speedBumpExpected(true)
+                    .build();
+            AadPromptHandler promptHandler = new AadPromptHandler(promptHandlerParameters);
+            promptHandler.handlePrompt(userName, password);
+        };
+
+        MsalAuthTestParams acquireTokenAuthParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(userName)
+                .scopes(Arrays.asList(getScopes()))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .authority(getAuthority())
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalSdk msalSdk = new MsalSdk();
+        // Acquire token interactively
+        MsalAuthResult acquireTokenResult = msalSdk.acquireTokenInteractive(acquireTokenAuthParams, interactionHandler, TokenRequestTimeout.SHORT);
+        Assert.assertFalse("Verify accessToken is not empty", TextUtils.isEmpty(acquireTokenResult.getAccessToken()));
+
+        // change the time on the device
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+
+        // Acquire token silently
+        MsalAuthResult acquireTokenSilentResult = msalSdk.acquireTokenSilent(acquireTokenAuthParams, TokenRequestTimeout.SHORT);
```

Comment:
Since we have removed the throttling at this time..a 30 second wait is not necessary..however, if we put back the throttling (in-memory caching) change back in, then we would need a 30 second wait here to truly test AT expiry because otherwise we would just be getting tokens from throttling cache....to be safe for the future, you may just want to put the 30 second sleep already there just in case we add back throttling in the future (which I do suspect we would do at some point)

Reply:
Couple of things
1. We are changing the device time to advance by 1 day here. I assume that should enable bypassing throttling cache, unless throttling cache gets its time from somewhere else than device time.
2. We should be be providing a setting to set the throttling time window, and should be able to override that from test to either 0 or 1 sec. Keeping a sleep of 30 sec in the test is going to increase the test execution time by 30 sec each time (multiplied by the number of tests using such sleep), which we should avoid.

Reply:
yes agreed....we should do everything we can to reduce test execution time. We do (did) have a way to clear the cache at least but the existing tests weren't doing that to just follow the behavior of actual code (but we can change that later)...I am not too sure about whether advancing the clock clears the throttling cache...we will have to test it (that means you'll have to put throttling change back in to test it)....since we don't have throttling in there right now..I am fine with you filing a work item to address these concerns (and anyone can pick it up later when we add throttling back)